platform/mellanox/mlnx-platform-api/sonic_platform/platform.py

```diff
@@ -22,15 +22,18 @@
 
 try:
     from sonic_platform_base.platform_base import PlatformBase
-    from .chassis import Chassis, ModularChassis
+    from .chassis import Chassis, ModularChassis, SmartSwitchChassis
     from .device_data import DeviceDataManager
 except ImportError as e:
     raise ImportError(str(e) + "- required module not found")
 
+
 class Platform(PlatformBase):
     def __init__(self):
         PlatformBase.__init__(self)
-        if DeviceDataManager.get_linecard_count() == 0:
+        if DeviceDataManager.get_dpu_count():
```

Comment:
is there a better way to do the check? get_dpu_count will read platform.json, it will slow down the speed of creating chassis object in other platform which does not support DPU.

Reply:
In the existing code we are reading the sysfs file first for existence of [line cards for modular chassis ](https://github.com/sonic-net/sonic-buildimage/blob/157818104e4a109421a72d4753b07a442c19e8be/platform/mellanox/mlnx-platform-api/sonic_platform/platform.py#L33)
Wouldn't it take similar amount of time for reading the platform.json? Because the platform.json is the source of truth to differentiate between smart switch and non smart switch platforms, I can check the time taken to read the platform.json file on average to double check if required

Reply:
reading a sysfs should be much faster than reading and parsing a json file